speech/use-cases/storytelling/storytelling.ipynb

```diff
@@ -0,0 +1,828 @@
+        "\n",
+        "        # If gender is indeterminate/androgynous, pick random one\n",
+        "        gender = choice(supported_genders) if gender not in supported_genders else gender\n",
+        "\n",
+        "        # Assign Voice to Character and don't reuse.\n",
+        "        voice = choice(gender_to_voices[gender])\n",
+        "        gender_to_voices[gender].remove(voice)\n",
+        "        character_to_voice[name] = voice\n",
+        "\n",
+        "    return character_to_voice\n",
+        "\n",
+        "\n",
+        "def synthesize_text(text: str, output: str, voice_name: str, language_code: str = DEFAULT_LANGUAGE):\n",
+        "    input_text = texttospeech.SynthesisInput(text=text)\n",
+        "    voice = texttospeech.VoiceSelectionParams(language_code=language_code, name=voice_name)\n",
+        "    audio_config = texttospeech.AudioConfig(audio_encoding=texttospeech.AudioEncoding.MP3)\n",
+        "\n",
+        "    response = tts_client.synthesize_speech(input=input_text, voice=voice, audio_config=audio_config)\n",
+        "\n",
+        "    # The response's audio_content is binary.\n",
+        "    with open(output, \"wb\") as f:\n",
+        "        f.write(response.audio_content)\n",
+        "\n",
+        "\n",
+        "def combine_audio_files(audio_files: List[str], filename: str) -> str:\n",
+        "    full_audio = AudioSegment.silent(duration=SILENCE_LENGTH)\n",
+        "\n",
+        "    for file in audio_files:\n",
+        "        sound = AudioSegment.from_mp3(file)\n",
+        "        silence = AudioSegment.silent(duration=SILENCE_LENGTH)\n",
+        "        full_audio += sound + silence\n",
+        "        os.remove(file)\n",
+        "\n",
+        "    outfile_name = f\"{Path(filename).stem}-complete.mp3\"\n",
+        "    full_audio.export(outfile_name, format=\"mp3\")\n",
+        "    return outfile_name\n",
+        "\n",
+        "\n",
+        "def get_characters(input_file: str) -> List[str]:\n",
+        "    character_list = []\n",
+        "    with open(input_file, \"r\") as f:\n",
+        "        lines = f.readlines()\n",
+        "\n",
+        "    start_line = lines.index(\"Characters:\\n\")\n",
+        "\n",
+        "    for i in range(start_line + 2, len(lines)):\n",
+        "        if lines[i] == \"\\n\":\n",
+        "            break\n",
+        "        character_list.append(lines[i].strip())\n",
+        "    return character_list\n",
+        "\n",
+        "\n",
+        "def parse_file(input_file: str, character_to_voice: Dict[str, Tuple[str, str]]) -> List[str]:\n",
+        "    with open(input_file, \"r\") as f:\n",
+        "        lines = f.readlines()\n",
+        "\n",
+        "    line_number = 1\n",
+        "    output_files = []\n",
+        "    filename = Path(input_file).stem\n",
+        "\n",
+        "    for line in lines:\n",
```

Comment:
Suggesting adding/changing the following lines to the notebook so that users will get a nice progress bar as it loops through the lines in the text file:

```python
%pip install --user --upgrade -q [...] tqdm

[...]

from tqdm import tqdm

[...]

for line in tqdm(lines, "Parsing input text"):
```

![Screenshot 2024-01-30 at 2 43 15 PM](https://github.com/GoogleCloudPlatform/generative-ai/assets/303236/9741df4c-3c82-4f4e-bdb8-1f71baaf4ba4)

Reply:
Good idea!